PR for keymap sigma-squared

Signynt opened the pull request:
Added a keymap for the bm40hsrgb called sigma-squared that is based off of the Miryoku layout. I asked manna-harbour for permission.

## Description

Added a Keymap

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on .gitignore:
```diff
@@ -1,79 +0,0 @@
-.history/
```
Please revert the changes to this file.

Signynt replied:
I reverted the file to what it was in the master, hopefully it's correct now

drashna commented on keyboards/bm40hsrgb/keymaps/sigma-squared/config.h:
```diff
@@ -0,0 +1,78 @@
+// qmk flash -kb bm40hsrgb -km sigma-squared
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Signynt replied:
Added it. Is it correct that I only need to add this to the config.h? Or is it necessary to add this to every file?

drashna replied:
To every `c` and `h` file. No need to add it to the makefile or readmes

drashna commented on keyboards/bm40hsrgb/keymaps/sigma-squared/keymap.c:
```diff
@@ -0,0 +1,108 @@
+// qmk flash -kb bm40hsrgb -km sigma-squared
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

drashna commented on keyboards/bm40hsrgb/keymaps/sigma-squared/sigma-squared.c:
```diff
@@ -0,0 +1,56 @@
+// qmk flash -kb bm40hsrgb -km sigma-squared
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

drashna commented on keyboards/bm40hsrgb/keymaps/sigma-squared/sigma-squared.h:
```diff
@@ -0,0 +1,20 @@
+// qmk flash -kb bm40hsrgb -km sigma-squared
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Signynt reviewed (commented):
Made requested changes

Signynt reviewed (commented):
added the header to the rest of the c and h files.

drashna reviewed (approved):
Looks good. 

Only thing left, is that we prefer that the keymap name match the github account name, as this makes tracking ownership easier, if a change is needed to be made to your keymap.

drashna commented on keyboards/bm40hsrgb/keymaps/sigma-squared/rgb_matrix_user.inc:
```diff
@@ -0,0 +1,440 @@
+// !!! DO NOT ADD #pragma once !!! //
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

spidey3 replied:
This is the only remaining issue that I see.

spidey3 replied:
This is the only remaining issue that I see.

spidey3 commented:
This looks ready to go, except for addressing the one remaining missing copyright notice in [rgb_matrix_user.inc](https://github.com/qmk/qmk_firmware/pull/11694/files/ae503035e6855da267d19a68392103ad018a51bf#diff-6ecf634bc6c2d074df167cdd47e41ff630634dd7f4afb822640a95476e672cf1).

Signynt commented:
I deleted the rgb file and added the code from the two extra .c and .h files to the keymap.c and config.h files.
The keymap has been renamed to signynt, to match my github username.

Signynt reviewed (commented):
changed keymap name as per request, all the files except for the rules.mk have the license header now.

Signynt reviewed (commented):
updated readme

drashna commented:
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!
